### PR TITLE
Minor slime enhancements

### PR DESCRIPTION
**Pull request by Delimusca**
things this does:
-you can dragdrop yourself onto mobs to feed, no need for that awful feed verb popup.
-attacking another slime will pull them off if they're feeding, and steal some of their mass otherwise.

mousedrop may have issues, tested on most mobs living and dead, seems to work fine.

### CODE REVIEW

**Comment by dannno** [comment]
NICE :+1:

File: code/modules/mob/living/carbon/slime/powers.dm [hunk 1]
```diff
@@ -226,4 +226,16 @@
...
\ No newline at end of file
+		src << "<i>I am not old enough to reproduce yet...</i>"
+
+/mob/living/carbon/slime/verb/SetSlimeName()
+	set name = "Set Name"
+	set category = "Slime"
+	set desc = "This will change your name."
+
+	var/OldSlimeName = name
+	var/NewSlimeName = copytext(sanitize(input(usr, "Put your gooey title here.", "Pick a name!", OldSlimeName)),1,MAX_NAME_LEN)
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/slime/powers.dm]: brb setting name to "MCTRAITOR IS ANTAG" when I spawn from a blob split

Comment (Delimusca) [inline code/modules/mob/living/carbon/slime/powers.dm]: how would you get a slime out of a blob, i'm confused

Comment (Deantwo) [inline code/modules/mob/living/carbon/slime/powers.dm]: Could also go to an intercom and yell it to the whole crew.
So your argument is kinda invalid.

If someone does this and people see it, they will (hopefully) report it to an admin and get the person punished.

Comment (Deantwo) [inline code/modules/mob/living/carbon/slime/powers.dm]: That is what we have game admins for Imgoofball. Silmes, golems and others can also talk through a turned on intercom/radio.

The code here is a little messy though.

Comment (Delimusca) [inline code/modules/mob/living/carbon/slime/powers.dm]: the name gets reset when you evolve/reproduce, so only being able to set it once would be pointless. i'm going to chop out the name stuff, probably bring it back in a better way when i finish rewriting slimes eventually.

Comment (Deantwo) [inline code/modules/mob/living/carbon/slime/powers.dm]: In that case, isn't it pointless?

**Comment by Delimusca** [comment]
self naming chopped out due to slime name update issues and bad code, back to badgering people wit h labellers.

File: code/modules/mob/living/carbon/slime/powers.dm (hunk 1 above)
Comment (Incoming5643) [inline code/modules/mob/living/carbon/slime/powers.dm]: Player slimes don't create more player slimes when they split anyways, that'd be Overpowered as hell

Comment (Iamgoofball) [inline code/modules/mob/living/carbon/slime/powers.dm]: huh, I thought they did

Comment (dannno) [inline code/modules/mob/living/carbon/slime/powers.dm]: I think they might have a chance, if there are ghosts around with the preference enabled. At least it used to be that way.

File: code/modules/mob/living/carbon/slime/slime.dm [hunk 2]
```diff
@@ -204,6 +204,12 @@
 	updatehealth()
 	return
 
+/mob/living/carbon/slime/MouseDrop(var/atom/movable/A as mob|obj)
+	if(isliving(A) && A != usr)
+		var/mob/living/Food = A
+		if(Food.Adjacent(usr) && !stat && Food.stat != DEAD) //messy
```
Comment (Miauw62) [inline code/modules/mob/living/carbon/slime/slime.dm]: replace this with canusetopic.

Comment (Delimusca) [inline code/modules/mob/living/carbon/slime/slime.dm]: i don't see where canuseTopic would fit here. do you want me to make slime/canuseTopic() with the adjacency and stat check?

Comment (Miauw62) [inline code/modules/mob/living/carbon/slime/slime.dm]: you can replace everything here except the food.stat thing with canusetopic. it's centralized so you dont have to replace every single stat check everywhere when you add something new.

Comment (Delimusca) [inline code/modules/mob/living/carbon/slime/slime.dm]: oh, i looked at it again and noticed the living/canusetopic. alright, i'll pop it in and see if it works.

Comment (Delimusca) [inline code/modules/mob/living/carbon/slime/slime.dm]: tested with this:
if(canUseTopic(Food, BE_CLOSE, NO_DEXTERY) && Food.stat != DEAD)
didn't seem to check for unconcious/dead, also could drag myself through windoors onto someone, that may have existed before though, i'll check that as well.

Comment (Miauw62) [inline code/modules/mob/living/carbon/slime/slime.dm]: so, what is happening with this now

Comment (Delimusca) [inline code/modules/mob/living/carbon/slime/slime.dm]: well unless i'm missing something i can't replace any of the three with canusetopic(), so it remains as is. if it's still a problem, i can close this and bring it back in a better form if i eventually get around to rewriting some slime procs.

**Pull request merged**